### PR TITLE
Waypoint: Remove module version

### PR DESCRIPTION
**Pull request by paladin-devops**
The version of a no-code module is being removed from Waypoint templates and add-ons. The pinned version for a no-code module in HCP TF will be used at provision time.

### CODE REVIEW

**Review by catsby — APPROVED** [review]
This is probably fine with the current state of the provider. In the future we'd want to add an additional field and deprecate the old one:

- https://developer.hashicorp.com/terraform/plugin/framework/deprecations#provider-attribute-removal

**Review by HenryEstberg — COMMENTED** [review]
This all looks good, like you mentioned in your corresponding Waypoint PR this will ofc need to wait until that is merged and the SDK is updated as well. Because that SDK update will likely have to be included here, I will hold off on approving officially until then.

**Review by HenryEstberg — APPROVED** [review]

**Review by jasonpilz — APPROVED** [review]

**Pull request merged**
